### PR TITLE
Convert more time functions to asm

### PR DESCRIPTION
**Pull request by colinleroy**
_time_t_to_tm: 46 bytes size gain, -8% cycles on the unit tests
mktime: 415 bytes gain, -39% cycles on the unit tests
asctime: 7 bytes gain

### CODE REVIEW

File: libsrc/common/_time_t_to_tm.s [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        sta     ptr4
+        lda     #$00
+        sta     ptr4+1
+
+        ; Clear TM buf while we have zero in A
+        ldx     #.sizeof(tm)-1
+:       sta     TM,x
+        dex
+        bne     :-
+
```
Comment (Compyx) [inline libsrc/common/_time_t_to_tm.s:43]: I assume you mean to clear all of `TM` here, in which case you'll need `bpl`, not `bne` here.

Comment (colinleroy) [reply libsrc/common/_time_t_to_tm.s:43]: Indeed!

**Comment by rofl0r** [comment]
i wonder if it's such a good idea to simply convert C functions to asm, as support for other CPUs might be upcoming (e.g. 65c816 for SNES), and then those will lack these routines. in musl libc, we have a .c version for all functions, but if a .s file of the same name exists in the selected arch directory, that file is used instead via some makefile magic.

**Comment by mrdudz** [comment]
> i wonder if it's such a good idea to simply convert C functions to asm, as support for other CPUs might be upcoming (e.g. 65c816 for SNES), and then those will lack these routines. in musl libc, we have a .c version for all functions, but if a .s file of the same name exists in the selected arch directory, that file is used instead via some makefile magic.

First of all the chance of that ever happening is small. Very small in fact. And if it ever happened, you'll have to rewrite large parts of the runtime and library anyway to make proper use of the 65816 - if it really can't be implemented in conditional assembly, like it is done now for the 6502 variants. And if you really need the C versions back, you can always fish them from the repo history. Not that it will ever happen :)

**Comment by rofl0r** [comment]
> Not that it will ever happen :)

ah, i was looking the other day on the SNES thread, and it appeared that the 65816 support was already (almost) complete in a forked repo.

**Comment by mrdudz** [comment]
> > Not that it will ever happen :)
> 
> ah, i was looking the other day on the SNES thread, and it appeared that the 65816 support was already (almost) complete in a forked repo.

I am not aware of anything like that (what "SNES thread"?)

**Comment by colinleroy** [comment]
The 65816 is a moot point, isn't it? It'll ingest 6502 assembly with no problem.

**Comment by rofl0r** [comment]
> I am not aware of anything like that (what "SNES thread"?)

https://github.com/cc65/cc65/discussions/1886#discussioncomment-6178521
https://github.com/cc65/cc65/issues/2137

> The 65816 is a moot point, isn't it? It'll ingest 6502 assembly with no problem.

i never coded for it, but i recall that the manual mentioned something that it needs to be explicitly switched to/from 16 to 8 bit mode. how much of a problem that is in practice, *shrug*

**Comment by mrdudz** [comment]
> > I am not aware of anything like that (what "SNES thread"?)
> 
> [#1886 (reply in thread)](https://github.com/cc65/cc65/discussions/1886#discussioncomment-6178521) #2137

Ok, five year old discussion, and a dead fork that noone even bothered to try since then. :)

> > The 65816 is a moot point, isn't it? It'll ingest 6502 assembly with no problem.
> 
> i never coded for it, but i recall that the manual mentioned something that it needs to be explicitly switched to/from 16 to 8 bit mode. how much of a problem that is in practice, _shrug_

Its no problem at all. IIRC it even starts in emulation mode, and will basically work like a 65C02 then.

**Comment by colinleroy** [comment]
Hi,
Is anything missing for this PR to be merged?

**Comment by mrdudz** [comment]
This one seems fine to merge, thanks!

**Pull request merged**